### PR TITLE
feat: Move My Create Report To Platform: Moved API call of expenses to platform

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at abbb5ca</samp>

This pull request migrates the `my-create-report` page and the related services and models to use the platform v1 expenses model and service. It also refactors some code, fixes some errors, and adds a new feature to fetch all expenses with pagination.

### <samp>🤖 Generated by Copilot at abbb5ca</samp>

> _`my-create-report`_
> _uses platform v1 expenses_
> _a spring migration_

### <samp>🤖 Generated by Copilot at abbb5ca</samp>

*  Add a new property `queryParams` to the `ExpensesQueryParams` interface to allow passing arbitrary query parameters to the expenses API ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-eba4df1821d93fac2802fab3ab1fa042a4db856e2c846d0c9c5527200a41111eR5))
*  Remove the empty constructor from the `ExpenseService` class to remove unnecessary code and follow the linting rules ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-9a808fcce64c2148898fe6d9e1416110d4c58e965864d9951d63a34ba7aaa370L9-L10))
*  Add a new method `getAllExpenses` to the `ExpensesService` class to provide a way to fetch all the expenses that match the given query parameters, using pagination and concatenation ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-43a6608834bfbe4963726f4de34dd6082b69581858020acfa23be1e61037d9f5R37-R54))
  *  Replacing the `app-expense-card` component with the `app-expense-card-v2` component in the page template ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e701d3a765409cabbd5c4367c7db4e695ea8863b9f4456848b6dae9580c04ecbL63-R67), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e701d3a765409cabbd5c4367c7db4e695ea8863b9f4456848b6dae9580c04ecbL72-R73))
  *  Adding the `Expense` model and the `ExpensesService` from the platform v1 module as imports to the page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L15-R16))
  *  Replacing the `Expense` type with the `PlatformExpense` type in the page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L24-R27))
  *  Adding the `ExpensesService` as a dependency injection to the page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794R61))
  *  Updating the logic for creating the first report, getting the transaction ids of the selected expenses, selecting and deselecting expenses, getting the report title and the selected total amount, fetching the ready to report expenses, and mapping and filtering the ready to report expenses, to use the new properties and methods of the platform v1 expenses model and service, such as `id`, `is_reimbursable`, `amount`, `getAllExpenses`, `report_id`, `state`, `spent_at`, and `is.null` ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L87-R91), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L118-R117), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L167-R169), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L203-R205), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L232-R234), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L241-R254))
  *  Adding a new method `checkShowDt` to the page component to provide a way to check whether to show the date of an expense based on the previous expense date ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794R265-R276))
  *  Removing extra commas in the page component to fix syntax errors and follow the linting rules ([link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L129-R128), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L141-R140), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L153-R152), [link](https://github.com/fylein/fyle-mobile-app/pull/2572/files?diff=unified&w=0#diff-e33a2b899a6e74f70b44155fb78ebb6e7719bfe5ff09a9a23561038648ac3794L160-R159))

## Clickup
https://app.clickup.com/t/86ctv3kkf

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 1]
```diff
@@ -34,6 +34,24 @@ export class ExpensesService {
       .pipe(map((expenses) => expenses.data));
   }
 
+  getAllExpenses(params: ExpensesQueryParams): Observable<Expense[]> {
```
Comment (OmkarJ13) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:37]: Can we modify method below to use this as well?

Basically getReportExpenses can also use getAllExpenses, and we can avoid duplication of the batching logic

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 2]
```diff
@@ -84,12 +86,9 @@ export class MyCreateReportPage implements OnInit {
...
         if (allReportsCount === 0) {
-          const etxns = this.readyToReportEtxns?.filter((etxn) => etxn.isSelected);
-          const txnIds = etxns.map((etxn) => etxn.tx_id);
-          const selectedTotalAmount = etxns.reduce(
-            (acc, obj) => acc + (obj.tx_skip_reimbursement ? 0 : obj.tx_amount),
-            0,
-          );
+          const etxns = this.readyToReportEtxns?.filter((etxn) => this.selectedElements.includes(etxn));
+          const txnIds = etxns.map((etxn) => etxn.id);
+          const selectedTotalAmount = etxns.reduce((acc, obj) => acc + (obj.is_reimbursable ? 0 : obj.amount), 0);
```
Comment (OmkarJ13) [inline src/app/fyle/my-create-report/my-create-report.page.ts:91]: Pls rename etxns and txns terminology to expense, in platform world we don't use the older terms

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 3]
```diff
@@ -200,10 +199,10 @@ export class MyCreateReportPage implements OnInit {
   }
 
   getReportTitle(): Subscription {
-    const txnIds = this.selectedElements.map((etxn) => etxn.tx_id);
+    const txnIds = this.selectedElements.map((etxn) => etxn.id);
     this.selectedTotalAmount = this.selectedElements.reduce(
-      (acc, obj) => acc + (obj.tx_skip_reimbursement ? 0 : obj.tx_amount),
-      0,
+      (acc, obj) => acc + (obj.is_reimbursable ? 1 : obj.amount),
```
Comment (OmkarJ13) [inline src/app/fyle/my-create-report/my-create-report.page.ts:204]: Is this correct? Shouldn't we be checking `!is_reimbursable`

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 4]
```diff
@@ -275,6 +262,18 @@ export class MyCreateReportPage implements OnInit {
     this.homeCurrency$ = this.currencyService.getHomeCurrency();
   }
 
+  checkShowDt(expense: PlatformExpense, i: number): boolean {
+    if (
+      i > 0 &&
+      expense.spent_at &&
+      this.readyToReportEtxns[i - 1].spent_at &&
+      expense.spent_at.toDateString() === this.readyToReportEtxns[i - 1].spent_at.toDateString()
```
Comment (OmkarJ13) [inline src/app/fyle/my-create-report/my-create-report.page.ts:270]: Pls use variables here to shorten this, long condition which can be hard to read

**Review by OmkarJ13 — CHANGES_REQUESTED** [review]
Added comments

File: src/app/core/services/platform/v1/spender/expenses.service.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/core/services/platform/v1/spender/expenses.service.ts:37]: Yes, was waiting for Arjun to merge his changes, will modify this so it is easier to test.

**Review by OmkarJ13 — APPROVED** [review]

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 5]
```diff
@@ -229,52 +213,54 @@ export class MyCreateReportPage implements OnInit {
...
-                  etxn.showDt = false;
-                }
-                etxn.isSelected = true;
-
-                if (this.selectedTxnIds.length > 0) {
-                  if (this.selectedTxnIds.indexOf(etxn.tx_id) === -1) {
-                    etxn.isSelected = false;
+              etxns.forEach((etxn) => {
+                if (this.selectedExpenseIDs.length > 0) {
+                  if (this.selectedExpenseIDs.indexOf(etxn.id) === -1) {
```
Comment (Chethan-Fyle) [inline src/app/fyle/my-create-report/my-create-report.page.ts:229]: `if (!this.selectedExpenseIDs.includes(etxn.id))`

File: src/app/fyle/my-create-report/my-create-report.page.ts [hunk 6]
```diff
@@ -229,52 +213,54 @@ export class MyCreateReportPage implements OnInit {
...
-        this.selectedElements = this.readyToReportEtxns;
+        this.readyToReportExpenses = res;
+        this.selectedElements = this.readyToReportExpenses;
         this.getReportTitle();
       });
 
     this.homeCurrency$ = this.currencyService.getHomeCurrency();
   }
 
+  checkShowDt(expense: PlatformExpense, i: number): boolean {
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/my-create-report/my-create-report.page.ts:250]: I am using this method in the template, using the index to check the previous items date

**Pull request merged**